src/ClonesWithImmutableArgs.sol

```diff
@@ -137,7 +140,12 @@ library ClonesWithImmutableArgs {
             }
             // solhint-disable-next-line no-inline-assembly
             assembly {
-                instance := create(0, ptr, creationSize)
+                if iszero(salt) {
```

Comment:
maybe instead of this check, we should create 2 functions `clone` as before  and `cloneDeterministic`

Reply:
Yeah, happy to make a new PR to work on this, unless @z0r0z is up for continuing on this?

Reply:
Also super noob question, I'm not very good at assembly. But would there be a way to write a `predictDeterministic` function that predicts the address it'll deploy to without actually deploying the contract? I know OZ's `Clones` have that too.